File: submissions/keyboard-row.py
# https://leetcode.com/problems/keyboard-row

class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        rows = [
            'zxcvbnm',
            'asdfghjkl',
            'qwertyuiop'
        ]
        result = []
        for word in words:
            letters = set([letter.lower() for letter in word])
            for row in rows:
                for letter in letters:
                    if letter not in row:
                        break
                else:
                    result.append(word)
        
        return result

        

File: submissions/keyboard-row.go
/*
 * @lc app=leetcode id=500 lang=golang
 *
 * [500] Keyboard Row
 */

// @lc code=start
import (
	"strings"
)

func findWords(words []string) []string {
	rows := []string{"qwertyuiopQWERTYUIOP", "asdfghjklASDFGHJKJL", "zxcvbnmZXCVBNM"}
	res := []string{}

outer:
	for _, word := range words {

		neededRow := -1

		for i, row := range rows {
			if strings.ContainsAny(row, string(word[0])) {
				neededRow = i
				break
			}
		}

		for _, letter := range word[1:] {
			if !strings.ContainsAny(rows[neededRow], string(letter)) {
				continue outer
			}
		}
		res = append(res, word)
	}

	return res
}

// @lc code=end

